opentech/apply/funds/templatetags/submission_tags.py

```diff
@@ -0,0 +1,27 @@
+import re
+
+from django import template
+from django.utils.safestring import mark_safe
+
+from opentech.apply.funds.models import ApplicationSubmission
+
+register = template.Library()
+
+
+@register.filter
+def submission_links(value):
+    regex = re.compile('[^\w]\#(\d+)[^\w]')
+    links = {}
+    for match in regex.finditer(value):
+        try:
+            submission = ApplicationSubmission.objects.get(id=match[1])
+        except ApplicationSubmission.DoesNotExist:
+            pass
+        else:
+            links[f'#{submission.id}'] = f'<a href="{submission.get_absolute_url()}">{submission.title} <span class="mid-grey-text">#{submission.id}</span></a>'
+
+    if links:
+        for sid, link in links.items():
+            value = value.replace(sid, link)
```

Comment:
this should probably be `re.sub()` because otherwise you risk replacing all `#1234` with the link to `#123`

e.g. sub pattern  `f'[^\w]({sid})[^\w]`

https://docs.python.org/3/library/re.html#re.sub

Reply:
Good catch, this is a very real possibility. Changed now and worth the performance loss.